src/MelodyParser.js

```diff
@@ -0,0 +1,173 @@
+const durations = require('./durations.js');
+const note = require('./note.js');
+
+class MelodyParser {
+
+    //interface method: song - text song, melody - MelodyQueue
+    parse(song, melody) {
+
+        //reset melody queue
+        melody.queue = [];
+
+        this.melody = melody;
+
+        //parse to melody
+        let lines = this.lines(song);
+        console.log('Preparing.');
+        this.prepareCommands(lines);
+        console.log('Building melody...');
+
+        melody = this.buildMelody();
+        console.log('Parsing complete...');
+
+    }
+
+    //split to lines and cleanup - rid of unwanted spaces
+    lines(sng) {
+        //reset parser
+        this.defines = [];
+        this.plays = [];
+        this.sounds = [];
+
+        //initial cleaning
+
+        // make single space separator
+        while (sng !== (sng = sng.replace(/  /g, ' '))) {};
+
+        sng = sng.replace(/\( /g, '(');
```

Comment:
`sng = sng.replace(/\s+([(),;])\s+/g, '$1');

Reply:
Beautiful! Seriously. One line instead several.

Reply:
example:
```js
> ' ( A ; B , C ; D ) '.replace(/\s+([(),;])\s+/g, '$1');
'(A;B,C;D)'
```

Reply:
Unfortunately this formula also doesn't work. Test failed. I stick with original replaces.

Reply:
:(